libs/agno/agno/vectordb/qdrant/qdrant.py

```diff
@@ -179,6 +229,31 @@ def insert(self, documents: List[Document], filters: Optional[Dict[str, Any]] =
             self.client.upsert(collection_name=self.collection, wait=False, points=points)
         log_debug(f"Upsert {len(points)} documents")
 
+    async def async_insert(self, documents: List[Document], filters: Optional[Dict[str, Any]] = None) -> None:
+        """Insert documents asynchronously."""
+        log_debug(f"Inserting {len(documents)} documents asynchronously")
+        points = []
+        for document in documents:
+            document.embed(embedder=self.embedder)
+            cleaned_content = document.content.replace("\x00", "\ufffd")
+            doc_id = md5(cleaned_content.encode()).hexdigest()
+            points.append(
+                models.PointStruct(
+                    id=doc_id,
+                    vector=document.embedding,
+                    payload={
+                        "name": document.name,
+                        "meta_data": document.meta_data,
+                        "content": cleaned_content,
+                        "usage": document.usage,
+                    },
+                )
+            )
+            log_debug(f"Inserted document asynchronously: {document.name} ({document.meta_data})")
+        if len(points) > 0:
+            await self.async_client.upsert(collection_name=self.collection, wait=False, points=points)
+        log_debug(f"Upserted {len(points)} documents asynchronously")
```

Comment:
Can we somehow make this a asyncio.gather? So that each document is done in parallel?

Reply:
ok good idea, just tried it. compared to synchronous processing its a boost of ~2x btw